### PR TITLE
FIX #9 - Now we can blank targets

### LINKED ISSUE #9: When blanking the `targets` config option, scrape jobs are not removed from Prometheus-K8s
When I do this, the `prometheus.yml` config in [Prometheus K8s](https://charmhub.io/prometheus-k8s) still contains a target while it should not:

```
juju deploy prometheus-scrape-target-k8s targets --channel=beta --config targets=127.0.0.1 --config labels=foo:bar,foo2:bar2

juju config targets targets=
```

### PR DESCRIPTION
**Pull request by Abuelodelanada**
This PR fix #9.

Steps to check the bug is fixed:

1. Deploy Prometheus: `juju deploy prometheus-k8s --channel beta`
2. Build this charm: `charmcraft pack`
3. Deploy it: `juju deploy ./*.charm targets --channel=beta --config targets=127.0.0.1 --config labels=foo:bar,foo2:bar2`
4. Relate both charms: `juju add-relation prometheus-k8s targets`
5. Check etc/prometheus/prometheus.yml:
```bash
root@prometheus-k8s-0:/etc/prometheus# cat prometheus.yml
global:
  evaluation_interval: 1m
  scrape_interval: 1m
  scrape_timeout: 10s
rule_files:
- /etc/prometheus/rules/juju_*.rules
scrape_configs:
- honor_timestamps: true
  job_name: prometheus
  metrics_path: /metrics
  scheme: http
  scrape_interval: 5s
  scrape_timeout: 5s
  static_configs:
  - targets:
    - localhost:9090
- honor_labels: true
  job_name: juju_observability_fb7d3c3_targets_external_jobs
  static_configs:
  - labels:
      foo: bar
      foo2: bar2
    targets:
    - 127.0.0.1
```
6. Blank target: `juju config targets targets=`
7. Re-check etc/prometheus/prometheus.yml and verify scrape jobs are removed:
```bash
root@prometheus-k8s-0:/etc/prometheus# cat prometheus.yml
global:
  evaluation_interval: 1m
  scrape_interval: 1m
  scrape_timeout: 10s
rule_files:
- /etc/prometheus/rules/juju_*.rules
scrape_configs:
- honor_timestamps: true
  job_name: prometheus
  metrics_path: /metrics
  scheme: http
  scrape_interval: 5s
  scrape_timeout: 5s
  static_configs:
  - targets:
    - localhost:9090
```

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -69,10 +69,11 @@ def _update_prometheus_jobs(self, _):
...
 
-        if jobs := self._scrape_jobs():
-            for relation in self.model.relations[self._prometheus_relation]:
-                relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+        jobs = self._scrape_jobs()
+        for relation in self.model.relations[self._prometheus_relation]:
+            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
 
+        if jobs:
             self.unit.status = ActiveStatus()
```
Comment (sed-i) [inline src/charm.py:77]: Should there be an `else` here? Otherwise after targets is blanked, status would remain as Active.

**Review by sed-i — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (Abuelodelanada) [reply src/charm.py:77]: Nop, after targets is blanked, status is `Blocked` (see line 101)

**Comment by balbirthomas** [comment]
This duplicates https://github.com/canonical/prometheus-scrape-target-k8s-operator/pull/12

**Comment by Abuelodelanada** [comment]
Follow this in #12

**Pull request closed without merging**